processing/src/main/java/org/apache/carbondata/processing/surrogatekeysgenerator/csvbased/CarbonCSVBasedSeqGenStep.java

```diff
@@ -994,6 +1006,10 @@ private String getCarbonLocalBaseStoreLocation() {
                 isStringDataType[j], byteBufferArr);
         if (!isSerialized && !isStringDataType[j] && CarbonCommonConstants.MEMBER_DEFAULT_VAL
             .equals(dimensionValue)) {
+          // if isDataLoadFail is set to true then no need to log in the bad record logger
+          if (badRecordsLogger.isDataLoadFail()) {
```

Comment:
no need to validate in multiple places rather please move this logic to method where it will decide to log the bad record details or make data load failed.